examples/tv-casting-app/tv-casting-common/core/CastingPlayerDiscovery.cpp

```diff
@@ -63,16 +63,59 @@ CHIP_ERROR CastingPlayerDiscovery::StartDiscovery(uint32_t deviceTypeFilter)
 
 CHIP_ERROR CastingPlayerDiscovery::StopDiscovery()
 {
-    ChipLogProgress(Discovery, "CastingPlayerDiscovery::StopDiscovery() called");
+    ChipLogProgress(Discovery, "CastingPlayerDiscovery::StopDiscovery() mCastingPlayers: %lu, mCastingPlayersInternal: %lu",
+                    mCastingPlayers.size(), mCastingPlayersInternal.size());
     VerifyOrReturnError(mState == DISCOVERY_RUNNING, CHIP_ERROR_INCORRECT_STATE);
     ReturnErrorOnFailure(mCommissionableNodeController.StopDiscovery());
 
+    // Clear mCastingPlayersInternal of disconnected CastingPlayers
+    ClearDisconnectedCastingPlayersInternal();
+    for (const auto & player : mCastingPlayers)
+    {
+        mCastingPlayersInternal.push_back(player);
+    }
+    // Clear mCastingPlayers of all CastingPlayers
     mCastingPlayers.clear();
     mState = DISCOVERY_READY;
 
     return CHIP_NO_ERROR;
 }
 
+void CastingPlayerDiscovery::ClearDisconnectedCastingPlayersInternal()
+{
+    ChipLogProgress(Discovery, "CastingPlayerDiscovery::ClearDisconnectedCastingPlayersInternal() mCastingPlayersInternal: %lu",
+                    mCastingPlayersInternal.size());
+    // Only clear the CastingPlayers in mCastingPlayersInternal with ConnectionState == CASTING_PLAYER_NOT_CONNECTED
+    ClearDisconnectedCastingPlayers(mCastingPlayersInternal);
+}
+
+void CastingPlayerDiscovery::ClearDisconnectedCastingPlayers(std::vector<std::shared_ptr<CastingPlayer>> & castingPlayers)
+{
```

Comment:
If ClearDisconnectedCastingPlayers() is only called from ClearDisconnectedCastingPlayersInternal(), can we just move its implementation to ClearDisconnectedCastingPlayersInternal()?

Reply:
That works.